test/integration/text_layer_spec.mjs

```diff
@@ -290,4 +290,44 @@ describe("Text layer", () => {
       });
     });
   });
+  describe("Text layer is aligned with the canvas layer", () => {
+    let pages;
+    beforeAll(async () => {
+      pages = await loadAndWait(
+        "Misaligned Text Layer.pdf",
+        `.page[data-page-number = "1"] .endOfContent`
+      );
+    });
+    afterAll(async () => {
+      await closePages(pages);
+    });
+
+    it("The width of text-span is correct", async () => {
+      await Promise.all(
+        pages.map(async ([browserName, page]) => {
+          const lines = [
+            "exceeding 1000 mm in Kyushu district, southwestern Japan, in",
+            "early July 2020. Especially, an elongated and stagnated mesoscale",
+            "convective system formed around the Kuma River in central",
+            "Kyushu district produced localized heavy rainfall with precipita-",
+            "tion amounts larger than 600 mm in 13 hours. Characteristics of",
+          ];
+          const expectedWidth = 227;
+          const expectedWidthChrome = 242;
+          // In Chrome, an extra space character is included at the end
+          // of the text-span resulting in difference in widths
```

Comment:
The difference in selection:
Chrome: 
![image](https://github.com/mozilla/pdf.js/assets/62544124/d607ae25-0b0d-4653-8a3a-c35efb92cb9f)

Firefox: 
<img width="361" alt="firefox" src="https://github.com/mozilla/pdf.js/assets/62544124/e8c6bff6-ec8c-4b07-8307-81d3a2af6ec7">